libmamba/src/core/pool.cpp

```diff
@@ -150,13 +159,146 @@ namespace mamba
         return id;
     }
 
-    std::optional<PackageInfo> MPool::id2pkginfo(Id id)
+    namespace
+    {
+        auto make_package_info(::Solvable* s) -> PackageInfo
```

Comment:
Looks like this should be a reference as we access `*s` directly without checking. I understand this function is just moved but we can do that change too while we're at it.